### ISSUE TITLE
Bad interaction of ANSI escapes and `\r`

### ISSUE THREAD

**Issue by gaborcsardi** [rstudio/rstudio#7649]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.735
    OS Version      : macOS Mojave
    R Version       : 4.0.2

### Steps to reproduce the problem

RUn this code:
```r
bug <- c(
  "⠹ [ \033[32mPASS\033[39m x632 \033[31mFAIL\033[39m x16 \033[35mWARN\033[39m x4",
  "\r", 
  "\033[32m✓\033[39m |   4       | reporter-zzz\033[36m [\033[0m"
)
cat(bug, sep = "")
```

### Describe the problem in detail

The ANSI colors and the `"\r" seem to interact badly, and there is some leftover text:

```
> cat(bug)
 ✓ |   4       | reporter-zzz4 
```

### Describe the behavior you expected

In a terminal this looks as

```
❯ cat(bug, sep = "")
✓ |   4       | reporter-zzz [
```

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
Investigation note: same problem in 1.2 and 1.3, so not a regression.

**Comment by gaborcsardi** [comment]
Great, thanks a lot!

**Comment by R-Hannibal** [comment]
@gaborcsardi, thanks for logging this and for the detailed issue report.

We have a fix in place in the latest Daily builds of 1.4 (available [HERE](https://dailies.rstudio.com/)). If you get the chance will you let us know if the issue is resolved for you in that build?

**Comment by gaborcsardi** [comment]
Thanks both, the fix seems to work really well. :)
